bot/exts/evergreen/youtube.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from dataclasses import dataclass
+from html import unescape
+from typing import List
+from urllib.parse import quote_plus
+
+from discord import Embed
+from discord.ext import commands
+from discord.utils import escape_markdown
+
+from bot.constants import Colours, Tokens
+
+log = logging.getLogger(__name__)
+
+KEY = Tokens.youtube
+SEARCH_API = "https://www.googleapis.com/youtube/v3/search"
+YOUTUBE_VIDEO_URL = "https://www.youtube.com/watch?v={id}"
+YOUTUBE_SEARCH_URL = "https://www.youtube.com/results?search_query={search}"
+RESULT = "`{index}` [{title}]({url}) - {author}"
+
+
+@dataclass
+class SearchResult:
+    """Represents one video."""
+
+    title: str
+    author: str
+    id: str
+
+
+class YouTubeSearch(commands.Cog):
+    """Sends the top 5 results of a query from YouTube."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    async def search_youtube(self, search: str) -> List[SearchResult]:
+        """Queries API for top 5 results matching the search term."""
+        results = []
+        async with self.http_session.get(
+            SEARCH_API,
+            params={"part": "snippet", "q": search, "type": "video", "key": KEY},
```

Comment:
Can we have this a constant variable which is being passed in here?

Reply:
@Shivansh-007  `search` isn't constant :)

Reply:
Hmmmm, yeah :(